src/libbson/CMakeLists.txt

```diff
@@ -248,10 +248,17 @@ set_target_properties (bson_shared PROPERTIES VERSION 0.0.0 SOVERSION 0)
 set_target_properties (bson_shared PROPERTIES OUTPUT_NAME "bson-${BSON_API_VERSION}" PREFIX "lib")
 
 if (ENABLE_APPLE_FRAMEWORK)
+
+   set (MONGOC_FRAMEWORK_BUNDLE_VERSION ${MONGOC_MAJOR_VERSION}.${MONGOC_MINOR_VERSION}.${MONGOC_MICRO_VERSION})
+   if ("${MONGOC_PRELEASE_VERSION}" STREQUAL "")
+      set (MONGOC_FRAMEWORK_BUNDLE_VERSION ${MONGOC_FRAMEWORK_BUNDLE_VERSION}${MONGOC_PRERELEASE_VERSION})
+   endif()
+
    set_target_properties(bson_shared PROPERTIES 
-      FRAMEWORK TRUE 
-      MACOSX_FRAMEWORK_BUNDLE_VERSION ${MONGOC_VERSION} 
-      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${MONGOC_VERSION}
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${BSON_API_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${MONGOC_FRAMEWORK_BUNDLE_VERSION}
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${MONGOC_RELEASED_VERSION}
```

Comment:
The short version string isn't allowed to be anything other than x.y.z, numeric. My understanding of MONGOC_RELEASED_VERSION is that it will, 1) always be an x.y.z version, and 2) will only differ meaningfully from MONGOC_VERSION when MONGOC_VERSION is in some prerelease state. That is exactly what we want to express here, I think: the short version string is the version that most closely models the actual three digit version we are, and if you need more detailed information (like that we are actually a prerelease of a newer version), then consult the BUNDLE_VERSION.

That would break down if there is ever a case where MONGOC_RELEASED_VERSION and MONGOC_VERSION are both just x.y.z but different.

Reply:
Ah okay, so when VERSION_RELEASED is "1.13.0" and VERSION_CURRENT is "1.14.0-prerelease" we want to set the bundle short string to 1.13.0.

Reply:
That is my best understanding, yes.